### PR TITLE
Add file name sanitization

### PR DESCRIPTION
**Pull request by optlsnd**
Added sanitizeFileName() function
Filename is being sanitized when the "add file name" option is enabled

### CODE REVIEW

File: widget/index.js [hunk 1]
```diff
@@ -30,6 +30,13 @@ function cropOption(mode, width, height) {
   }
 }
 
+function sanitizeFileName(fileName) {
+  var regexp = /[^A-Za-z0-9_]+/g
+  var extension = fileName.split('.').pop()
+  var name = fileName.substring(0, fileName.length - extension.length)
+  return name.replace(regexp, '') + '.' + extension.replace(regexp, '')
+}
```
Comment (Zmoki) [inline widget/index.js:43]: if a name will contain non-english words only, then the sanitized name will be empty, for example

```
sanitizeFileName('Снимок экрана.png')
```

result - `.png`

Our upload servers set name `noroot` in that cases

**Review by Zmoki — CHANGES_REQUESTED** [review]

File: widget/index.js (hunk 1 above)
Comment (optlsnd) [reply widget/index.js:43]: I've updated the function, now it checks the length of the sanitized name and returns 'noroot' if it's zero

Comment (Zmoki) [reply widget/index.js:43]: Sorry, we need to get `noroot` **plus extension**, 

for example

```
sanitizeFileName('Снимок экрана.png') // noroot.png
```

**Review by Zmoki — CHANGES_REQUESTED** [review]

File: widget/index.js (hunk 1 above)
Comment (optlsnd) [reply widget/index.js:43]: Fixed

**Review by Zmoki — APPROVED** [review]

**Pull request merged**